node/pkg/accounting/submit_obs.go

```diff
@@ -0,0 +1,275 @@
+	txResp, err := SubmitObservationToContract(acct.ctx, acct.logger, acct.gk, gsIndex, acct.wormchainConn, acct.contract, msg)
+	if err != nil {
+		acct.logger.Error("acct: failed to submit observation request", zap.String("msgId", msgId), zap.Error(err))
+		submitFailures.Inc()
+		return
+	}
+
+	alreadyCommitted, err := CheckSubmitObservationResult(txResp)
+	if err != nil {
+		submitFailures.Inc()
+		if strings.Contains(err.Error(), "insufficient balance") {
+			balanceErrors.Inc()
+			acct.logger.Error("acct: insufficient balance error detected, dropping transfer", zap.String("msgId", msgId), zap.Error(err))
+			acct.pendingTransfersLock.Lock()
+			defer acct.pendingTransfersLock.Unlock()
+			acct.deletePendingTransfer(msgId)
+		} else {
+			acct.logger.Error("acct: failed to submit observation request", zap.String("msgId", msgId), zap.Error(err))
+		}
+		return
+	}
+
+	if alreadyCommitted {
+		acct.pendingTransfersLock.Lock()
+		defer acct.pendingTransfersLock.Unlock()
+		pe, exists := acct.pendingTransfers[msgId]
+		if exists {
+			acct.logger.Info("acct: transfer has already been committed, publishing it", zap.String("msgId", msgId))
+			acct.publishTransfer(pe)
+			transfersApproved.Inc()
+		} else {
+			acct.logger.Debug("acct: transfer has already been committed, and it is no longer in our map", zap.String("msgId", msgId))
+		}
+	}
+}
+
+// SubmitObservationToContract is a free function to make a call to the smart contract to submit an observation request.
+func SubmitObservationToContract(
+	ctx context.Context,
+	logger *zap.Logger,
+	gk *ecdsa.PrivateKey,
+	gsIndex uint32,
+	wormchainConn *wormconn.ClientConn,
+	contract string,
+	msg *common.MessagePublication,
+) (*sdktx.BroadcastTxResponse, error) {
+	txHashStr := msg.TxHash.String()
+	if strings.Index(txHashStr, "0x") == 0 {
+		txHashStr = txHashStr[2:]
+	}
+	txHashBytes, err := hex.DecodeString(txHashStr)
+	if err != nil {
+		return nil, fmt.Errorf("acct: failed to decode tx_hash '%s': %w", msg.TxHash.String(), err)
+	}
+	obs := []Observation{
+		Observation{
+			TxHash:           base64.StdEncoding.EncodeToString(txHashBytes),
+			Timestamp:        uint32(msg.Timestamp.Unix()),
+			Nonce:            msg.Nonce,
+			EmitterChain:     uint16(msg.EmitterChain),
+			EmitterAddress:   msg.EmitterAddress,
+			Sequence:         msg.Sequence,
+			ConsistencyLevel: msg.ConsistencyLevel,
+			Payload:          base64.StdEncoding.EncodeToString(msg.Payload),
+		},
+	}
+
+	bytes, err := json.Marshal(obs)
+	if err != nil {
+		return nil, fmt.Errorf("acct: failed to marshal accounting observation request: %w", err)
+	}
+
+	b64String := base64.StdEncoding.EncodeToString(bytes)
+
+	digest := vaa.SigningMsg(bytes)
+
+	SignatureBytes, err := ethCrypto.Sign(digest.Bytes(), gk)
+	if err != nil {
+		return nil, fmt.Errorf("acct: failed to sign accounting Observation request: %w", err)
+	}
+
+	sig := SignatureType{Index: 0, Signature: SignatureBytes}
```

Comment:
should this always be 0?

Reply:
It most definitely should not! Good catch!